autoscaler/controllers/datacollection/daemonset.go

```diff
@@ -91,7 +92,8 @@ func syncDaemonSet(ctx context.Context, dests *odigosv1.DestinationList, datacol
 		return nil, err
 	}
 
-	desiredDs, err := getDesiredDaemonSet(datacollection, configMap.String(), scheme, imagePullSecrets, odigosVersion, odigletDaemonsetPodSpec)
```

Comment:
here is the original issue, we used `configMap.String()` which includes metadata that might change even when the content didn't.